src/varnam_state.cpp

```diff
@@ -403,12 +414,17 @@ void VarnamState::commitText(const FcitxKeySym &key) {
     reset();
     return;
   }
+    
 #ifdef DEBUG_MODE
   VARNAM_INFO() << "learn word:" << stringToCommit;
 #endif
   std::string wordToLearn(stringToCommit); // [TODO] try unique_ptr<char[]>
   std::thread learnThread(varnam_learn_word, engine_->getVarnamHandle(),
                           std::move(wordToLearn), 0);
+    if (isWordBreakKey) {
+        // Remove the word break character for learning
+        wordToLearn = wordToLearn.substr(0, wordToLearn.size() - 1);
```

Comment:
Instead of this you can initialize `wordToLearn` with `stringToCommit` value before adding the punctuation char.

Reply:
i will test this.

Reply:
Please make this change as well